domain-models-runtime/src/main/java/org/folio/rest/persist/Conn.java

```diff
@@ -866,4 +868,88 @@ static String durationMsg(String descriptionKey, String sql, long startNanoTime)
 
     return streamGet(table, clazz, PostgresClient.DEFAULT_JSONB_FIELD_NAME, wrapper, false, null, null, streamHandler);
   }
+
+  /**
+   * Send a parameterized/prepared statement.
+   *
+   * @param sql - the SQL command to run
+   * @param params - the values for the {@code $} placeholders, empty if none
+   * @return the reply from the database
+   */
+  public Future<RowSet<Row>> execute(String sql, Tuple params) {
+    try {
+      long start = log.isDebugEnabled() ? System.nanoTime() : 0;
+      // more than optimization.. preparedQuery does not work for multiple SQL statements
+      if (params.size() == 0) {
+        return pgConnection.query(sql).execute()
+            .onComplete(x -> log.debug(() -> durationMsg("execute", sql, start)));
+      }
+      return pgConnection.preparedQuery(sql).execute(params)
+          .onComplete(x -> log.debug(() -> durationMsg("execute", sql, start)));
+    } catch (Exception e) {
+      log.error(e.getMessage(), e);
+      return Future.failedFuture(e);
+    }
+  }
+
+  /**
+   * Send an SQL statement.
+   *
+   * <p>Use {@link #execute(String, Tuple)} with {@code $1, $2, ...} parameters to
+   * avoid SQL injection.
+   *
+   * @param sql - the SQL command to run
+   * @return the reply from the database
+   */
+  public Future<RowSet<Row>> execute(String sql) {
+    return execute(sql, Tuple.tuple());
+  }
+
+  /**
+   * Run a parameterized/prepared statement and
+   * run it with a list of sets of parameters.
+   *
+   * @param sql - the SQL command to run
+   * @param params - there is one list entry for each SQL invocation containing the parameters
+   *                    {@code $} placeholders.
+   * @return the reply from the database
+   */
+  public Future<List<RowSet<Row>>> execute(String sql, List<Tuple> params) {
+
+    try {
+      long start = log.isDebugEnabled() ? System.nanoTime() : 0;
+      Promise<List<RowSet<Row>>> promise = Promise.promise();
+      List<RowSet<Row>> results = new ArrayList<>(params.size());
+      Iterator<Tuple> iterator = params.iterator();
+      PreparedQuery<RowSet<Row>> preparedQuery = pgConnection.preparedQuery(sql);
+      Runnable task = new Runnable() {
```

Comment:
Wondering if `executeBatch` is not used here is because sequential execution is desired. If so, maybe the method name and doc can be more specific about that.

Reply:
Good catch!
I've created https://issues.folio.org/browse/RMB-819 for this change.